routes/merged/project/location/vm.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+class Clover
+  branch = lambda do |r|
+    r.get api? do
+      vm_list_api_response(vm_list_dataset)
+    end
+
+    r.on NAME_OR_UBID do |vm_name, vm_ubid|
+      if vm_name
+        r.post api? do
+          vm_post(vm_name)
+        end
+
+        filter = {Sequel[:vm][:name] => vm_name}
+      else
+        filter = {Sequel[:vm][:id] => UBID.to_uuid(vm_ubid)}
+      end
+
+      filter[:location] = @location
+      vm = @project.vms_dataset.first(filter)
+
+      unless vm
+        response.status = r.delete? ? 204 : 404
+        r.halt
+      end
+
+      r.get true do
+        Authorization.authorize(current_account.id, "Vm:view", vm.id)
+        @vm = Serializers::Vm.serialize(vm, {detailed: true})
+        if api?
+          @vm
+        else
+          view "vm/show"
+        end
```

Comment:
General Ruby styling question: What's your opinion on using ternary for single-line codes? Are they popular in Ruby?

```suggestion
        api? ? @vm : view "vm/show"
```

Reply:
I'm fine with using ternary operators.